extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/GraphiteEmitter.java

```diff
@@ -0,0 +1,187 @@
+  private final GraphiteEmitterConfig graphiteEmitterConfig;
+  private final AtomicBoolean started = new AtomicBoolean(false);
+  private final ObjectMapper mapper;
+  private final LinkedBlockingQueue<GraphiteEvent> eventsQueue;
+  private final static long DEFAULT_PUT_GET_TIMEOUT = 1000; // default wait for put/get operations on the queue 1 sec
+  private static final long FLUSH_TIMEOUT = 60000; // default flush wait 1 min
+  private final ScheduledExecutorService exec = Executors.newScheduledThreadPool(
+      2,
+      new ThreadFactoryBuilder().setDaemon(
+          true)
+                                .setNameFormat(
+                                    "GraphiteEmitter-%s")
+                                .build()
+  ); // Thread pool of two in order to schedule flush runnable
+
+  public GraphiteEmitter(@NotNull GraphiteEmitterConfig graphiteEmitterConfig, @NotNull ObjectMapper mapper)
+  {
+    this.mapper = mapper;
+    this.graphiteEmitterConfig = graphiteEmitterConfig;
+    this.graphiteEventConverter = new GraphiteEventConverter(
+        graphiteEmitterConfig.getNamespacePrefix(),
+        graphiteEmitterConfig.getIgnoreHostname(),
+        graphiteEmitterConfig.getIgnoreServiceName(),
+        graphiteEmitterConfig.getWhiteListDimsMapper()
+    );
+    this.eventsQueue = new LinkedBlockingQueue(graphiteEmitterConfig.getMaxQueueSize());
+  }
+
+  @Override
+  public void start()
+  {
+    log.info("Starting Graphite Emitter.");
+    synchronized (started) {
+      if (!started.get()) {
+        exec.scheduleAtFixedRate(
+            new ConsumerRunnable(),
+            graphiteEmitterConfig.getFlushPeriod(),
+            graphiteEmitterConfig.getFlushPeriod(),
+            TimeUnit.MILLISECONDS
+        );
+        started.set(true);
+      }
+    }
+  }
+
+
+  @Override
+  public void emit(Event event)
+  {
+    if (!started.get()) {
+      throw new ISE("WTF emit was called while service is not started yet");
+    }
+
+    if (event instanceof ServiceMetricEvent) {
+      final ServiceMetricEvent serviceMetricEvent = (ServiceMetricEvent) event;
+      final GraphiteEvent graphiteEvent = graphiteEventConverter.druidMetricToGraphite(serviceMetricEvent);
+      if (graphiteEvent == null) {
+        return;
+      }
+      try {
+        final boolean isSuccessful = eventsQueue.offer(graphiteEvent, DEFAULT_PUT_GET_TIMEOUT, TimeUnit.MILLISECONDS);
+        if (!isSuccessful) {
+          log.error(
+              "Throwing event [%s] on the floor Graphite queue is full please increase the capacity or/and the consumer frequency",
+              mapper.writeValueAsString(event)
+          );
+        }
+      }
+      catch (InterruptedException | JsonProcessingException e) {
+        Thread.interrupted();
+        log.error(e, e.getMessage(),Thread.interrupted());
+      }
+    } else if (graphiteEmitterConfig.getLogAlerts() && event instanceof AlertEvent) {
+      try {
+        log.warn(mapper.writeValueAsString(event));
```

Comment:
@himanshug If we ask a user to define a composing emitter every time they want to use the graphite emitter,it would be little overkill for the end user and also can lead to scenarios where  user only configures an graphite emitter and hence loose the alert events.
In my mind,a better design would have been to extend GraphiteEmitter from Composing emitter such that the first emitter is graphite and others are user defined and in case not defined by user default is LoggingEmitter.
In this case we would get the best of both i.e user will need to worry about defining other emitters only if they want the alert to go somewhere else in default case also everything works just as good.